File: example/lib/print_tool/page.dart
import 'dart:typed_data';

import './line.dart';

enum PageWidth {
  p58,
  p80,
}

extension PageWidthExOnInt on int {
  PageWidth? toPageWidth() {
    switch (this) {
      case 58:
        return PageWidth.p58;
      case 80:
        return PageWidth.p80;
      default:
        return null;
    }
  }
}

class Page {
  final PageWidth pageWidth;
  final List<Line> _lines = [];
  Page({required this.pageWidth});
  Uint8List build() {
    List<int> content = [];
    for (var line in _lines) {
      content.addAll(line.build(pageWidth));
    }
    return Uint8List.fromList(content);
  }

  void addLine(Line line) {
    _lines.add(line);
  }

  void addLines(List<Line> lines) {
    _lines.addAll(lines);
  }
}

File: src/main/java/com/printer/Page.java
package com.printer;

import java.util.ArrayList;
import java.util.List;

public class Page {
    private final PageWidth pageWidth;
    private final List<Line> lines;

    public Page(PageWidth pageWidth) {
        this.pageWidth = pageWidth;
        this.lines = new ArrayList<>();
    }

    public byte[] build() {
        List<Byte> content = new ArrayList<>();
        for (Line line : lines) {
            byte[] lineBytes = line.build(pageWidth);
            for (byte b : lineBytes) {
                content.add(b);
            }
        }
        byte[] result = new byte[content.size()];
        for (int i = 0; i < content.size(); i++) {
            result[i] = content.get(i);
        }
        return result;
    }

    public void addLine(Line line) {
        lines.add(line);
    }

    public void addLines(List<Line> lines) {
        this.lines.addAll(lines);
    }
} 